torch/csrc/jit/mobile/model_compatibility.h

```diff
@@ -20,5 +21,20 @@ TORCH_API int64_t _get_model_bytecode_version(const std::string& filename);
 TORCH_API int64_t _get_model_bytecode_version(
     std::shared_ptr<caffe2::serialize::ReadAdapterInterface> rai);
 
+// Struct storing metadata of an operator that can be useful for versioning
+struct OperatorInfo {
+  int num_args;
+};
+
+// The family of methods below to get the root ops and information from a model
```

Comment:
No by root ops its only means the top level ops. Theres no way to get what we call the 'traced' operators on a model without running the tracer. Thats the problem I was referring too in the design doc when I mentioned that getting the model ops seems difficult to me. If we want to get the full op list and not just the root ops we will need to come up with a different solution that loads the model up and traces it. cc @iseeyuan

Reply:
That's right. This (and the need for tracing) is the unfortunate consequence of TS IR being ignorant of the compositions done at the dispatcher level.

I suggest we document a couple of things:
1. What you mention that root ops are only provided (as you did) and that traced ops are not worth to add since it requires the current runtime as input (and consequently its current transient dependencies). 
2. The corner case for potential false positives: The model-runtime compatibility check done by validating the model-level root ops (returned by this API) are a subset of all ops (root and transient) available in a runtime (returned by the runtime API), can incorrectly pass even when the root ops in the model have transient op dependencies not present in the runtime.

I suggest we should create a "manual" type doc for these APIs that document how to use, and these caveats.

@iseeyuan

Reply:
There is possibility of false positive on "operator subset" check. 
For other compatibility check, such as schema and versioning, it should be safer: 
If a traced op is updated, the caller ops (for example, the root ops) should be responsible for this update. So checking the root ops only would be fine. 
But for subset check there's a gap between the static model and dynamic runtime behavior. Hopefully TS IR could include those compositions including dtype, to have this check more complete.